API - List restaurants endpoint(index)

MikaelFeher opened the pull request:
PT Story: https://www.pivotaltracker.com/story/show/129453853

Changes proposed in this pull request:
- Adds an endpoint to retrieve an index of restaurants in the database (indexed by name)

What I have learned working on this feature:
- A beginning view of the complexities of building an API.
- The importance of `let` vs `let!`.
- A beginning understanding how to set up jbuilder template.

Screenshots:
View from Postman:
![screen shot 2016-09-02 at 15 29 19 2](https://cloud.githubusercontent.com/assets/18580001/18205400/15c8031c-7122-11e6-9c17-b377a2716b21.png)
Localhost:
![screen shot 2016-09-02 at 15 30 57 2](https://cloud.githubusercontent.com/assets/18580001/18205450/53987ba4-7122-11e6-9c70-589d558a8aae.png)

Ready for review!

MikaelFeher committed:
Adds request test for retrieving a restaurant
Adds request test for returning a list

MikaelFeher committed:
Minor refactoring

MikaelFeher committed:
Updates controller to include jbuilder view
Adds jbuilder formatting and view

MikaelFeher committed:
Merge branch 'develop' of github.com:CraftAcademy/slow_food_online_june into api_rest_index

tochman commented on config/routes.rb:
```diff
@@ -16,5 +16,8 @@
       resources :ping, only: [:index], constraints: {format: /(json)/}
     end
+    namespace :v1 do
+      resources :restaurants, only: [:index] #, constraints: {format: /(json)/}
```
Why is the json constraint commented out?

tochman commented on app/views/v1/restaurants/index.json.jbuilder:
```diff
@@ -0,0 +1,3 @@
+json.restaurants @restaurants do |restaurant|
+  json.name restaurant.name
```
Is the restaurant name the only thing we want this endpoint to expose?

MikaelFeher committed:
Extends the information that is indexed
geocoding set to nil in the tests due to implementation issues
that will be taken care of as a separate issue
